sound/soc/sof/intel/pci-ptl.c

```diff
@@ -55,10 +55,40 @@ static const struct sof_dev_desc ptl_desc = {
 	.ops_init = sof_ptl_ops_init,
 };
 
+static const struct sof_dev_desc wcl_desc = {
+	.use_acpi_target_states	= true,
+	.machines               = snd_soc_acpi_intel_ptl_machines,
+	.alt_machines		= snd_soc_acpi_intel_ptl_sdw_machines,
+	.resindex_lpe_base      = 0,
+	.resindex_pcicfg_base   = -1,
+	.resindex_imr_base      = -1,
+	.irqindex_host_ipc      = -1,
+	.chip_info		= &wcl_chip_info,
+	.ipc_supported_mask	= BIT(SOF_IPC_TYPE_4),
+	.ipc_default		= SOF_IPC_TYPE_4,
+	.dspless_mode_supported	= true,
+	.default_fw_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/wcl",
+	},
+	.default_lib_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/wcl",
+	},
+	.default_tplg_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-tplg",
+	},
+	.default_fw_filename = {
+		[SOF_IPC_TYPE_4] = "sof-wcl.ri",
+	},
+	.nocodec_tplg_filename = "sof-wcl-nocodec.tplg",
```

Comment:
How should this be?
we use the `snd_soc_acpi_intel_ptl_machines` and `snd_soc_acpi_intel_ptl_sdw_machines` but the nocodec is using `sof-wcl-`
Should we duplicate the ptl mach with s/ptl/wcl applied or change the nocodec to sof-ptl as well?

WCL is in essence PTL with reduced number of cores.

Reply:
I think in general WCL tplgs should be separate. Let's have nocodec separate as well. sof-ptl-nocodec.tplg only uses cores 0+1+2 now, but if this is changed at some point, it'll break WCL machines using this (albeit this is for R&D but still).